tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -325,6 +325,80 @@ def testee(a: cases.IFloatField) -> gtx.Field[[IDim], int64]:
     )
 
 
+@pytest.mark.uses_tuple_returns
+def test_astype_on_tuples(cartesian_case):  # noqa: F811 # fixtures
+    @gtx.field_operator
+    def field_op_returning_a_tuple(
+        a: cases.IFloatField, b: cases.IFloatField
+    ) -> tuple[gtx.Field[[IDim], float], gtx.Field[[IDim], float]]:
+        tup = (a, b)
+        return tup
+
+    @gtx.field_operator
+    def cast_tuple(
+        a: cases.IFloatField,
+        b: cases.IFloatField,
+        a_casted_to_int_outside_of_gt4py: cases.IField,
+        b_casted_to_int_outside_of_gt4py: cases.IField,
+    ) -> tuple[gtx.Field[[IDim], bool], gtx.Field[[IDim], bool]]:
+        result = astype(field_op_returning_a_tuple(a, b), int32)
+        return (
+            result[0] == a_casted_to_int_outside_of_gt4py,
+            result[1] == b_casted_to_int_outside_of_gt4py,
+        )
+
+    @gtx.field_operator
+    def cast_nested_tuple(
+        a: cases.IFloatField,
+        b: cases.IFloatField,
+        a_casted_to_int_outside_of_gt4py: cases.IField,
+        b_casted_to_int_outside_of_gt4py: cases.IField,
+    ) -> tuple[gtx.Field[[IDim], bool], gtx.Field[[IDim], bool], gtx.Field[[IDim], bool]]:
+        result = astype((a, field_op_returning_a_tuple(a, b)), int32)
+        return (
+            result[0] == a_casted_to_int_outside_of_gt4py,
+            result[1][0] == a_casted_to_int_outside_of_gt4py,
+            result[1][1] == b_casted_to_int_outside_of_gt4py,
+        )
+
+    a = cases.allocate(cartesian_case, cast_tuple, "a")()
+    b = cases.allocate(cartesian_case, cast_tuple, "b")()
+    a_casted_to_int_outside_of_gt4py = cases.allocate(
+        cartesian_case, cast_tuple, "a", dtype=int32
+    )()
+    b_casted_to_int_outside_of_gt4py = cases.allocate(
+        cartesian_case, cast_tuple, "b", dtype=int32
+    )()
```

Comment:
```suggestion
    a_casted_to_int_outside_of_gt4py = gtx.np_as_located_field(np.asarray(a).astype(int32))
    b_casted_to_int_outside_of_gt4py = gtx.np_as_located_field(np.asarray(b).astype(int32))
```
This avoids relying on the `cases` utility to generate the same values (which only works because you specify the name of the argument). This is behavior we shouldn't rely on.